profile-widget/diveeventitem.cpp

```diff
@@ -176,10 +178,12 @@ void DiveEventItem::setupToolTipString(struct gasmix *lastgasmix)
 			name += tr(" (cyl. %1)").arg(internalEvent->gas.index + 1);
 		bool icd = isobaric_counterdiffusion(lastgasmix, mix, &icd_data);
 		if (icd_data.dHe < 0) {
-			if (icd)
-				name += tr("\nICD: ΔHe=%1% ΔN₂=%2%>%3%").arg(icd_data.dHe / 10).arg(icd_data.dN2 / 10).arg(-icd_data.dHe / 50);
-			else
-				name += tr("\nICD: ΔHe=%1% ΔN₂=%2%<%3%").arg(icd_data.dHe / 10).arg(icd_data.dN2 / 10).arg(-icd_data.dHe / 50);
+			put_format(&mb, "\n%s: %s=%+.3g%% %s=%+.3g%%%s%+.3g%%",
+				tr("ICD").toUtf8().constData(),
+				tr("ΔHe").toUtf8().constData(), icd_data.dHe / 10.0,
+				tr("ΔN₂").toUtf8().constData(), icd_data.dN2 / 10.0,
+				icd ? ">" : "<", -icd_data.dHe / 50.0);
+			name += QString::fromUtf8(mb.buffer, mb.len);
```

Comment:
@sfuchs79 @atdotde 
`put_format()` uses `realloc()`, which means that the buffer needs to be freed.

Reply:
Thanks for the hint. I will fix this together with another small issue I found.
Plus will wait for the final decision about correct precision for the % values.